src/project_rpg/SkillToken.java

```diff
@@ -56,6 +87,10 @@ private void nothing() {
 
   @Override
   public void run() {
+    if (skill.getBehavior().equals("earthAttack")) {
+      getGrid().getPlayer().reduceMana(skill.getCost());
```

Comment:
This line of code is repeated 4 times, which indicates that it can be reused. Move it before the `if` statement.